File: src/main/scala/fds/scala/queue/Producer.scala
package fds.scala.queue

import com.zink.queue.{WriteChannel, ConnectionFactory, Connection}

/**
  * Producer stub
  *
  * From Fly Docker
  *
  * > docker run -d -p 4396:4396 zink/fly
  *
  * To find container <run time name> (not required for running example)
  * > docker ps
  *
  * To find ipAddress (not required to run; point to localhost)
  * > docker inspect <run time name> | grep -i ipaddress
  *
  */
object Producer extends App {

  val con: Connection = ConnectionFactory.connect("127.0.0.1")
  val wc: WriteChannel = con.publish(ChannelName)

  wc.write("Hello Consumer")
  wc.write(EndOfStreamMarker)

  val fileName = "access.log"
  val logLines = scala.io.Source.fromFile(fileName).getLines

  // TODO use logLines to write each line onto the channel
//  for (line <- logLines) {
//    wc.write(line)
//  }
  // TODO write an end of stream marker
  wc.write(EndOfStreamMarker)

}

File: src/main/java/fds/java/queue/Producer.java
package fds.java.queue;


import com.zink.queue.Connection;
import com.zink.queue.ConnectionFactory;
import com.zink.queue.WriteChannel;

import java.io.FileInputStream;
import java.util.Scanner;


/**
 * Stub for the Producer of information to queue
 * <p>
 * From Fly Docker
 * <p>
 * > docker run -p 4396:4396 zink/fly
 * <p>
 * To check it is running
 * > docker ps
 */
public class Producer {

    public static void main(String[] args) throws Exception {

        final String ipAddr = "192.168.1.84";
        Connection con = ConnectionFactory.connect(ipAddr);
        WriteChannel wc = con.publish("BBC7");

        wc.write("Hello Consumer");

        /* Read a file from the local file system line by line */
        final String fileName = "file.log";
        FileInputStream fis = new FileInputStream(fileName);
        Scanner scanner = new Scanner(fis);
        while (scanner.hasNextLine()) {
            String line = scanner.nextLine();
            // TODO
            // Write to channel until the file is finished
        }
        // TODO
        // Write an end of stream marker
    }

}
